HomeTask5/MyNUnit/TestRunner.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        /// <summary>
+        /// Invoking Methods with MyTestAttribute
+        /// </summary>
+        public static void MethodsWithMyTestInvoker(MethodInfo methodInfo, object obj)
+        {
+            obj = ConstuctorFinder(methodInfo);
+            MyTestAttribute attribute = (MyTestAttribute)methodInfo.GetCustomAttribute(typeof(MyTestAttribute), true);
+            if (attribute.Ignore != null)
+            {
+                MyTests.Add(new TestStrcuct(methodInfo, isIgnored: true, ignore_message: attribute.Ignore));
+            }
+            else
+            {
+                if (attribute.Expected == null)
+                {
+                    MethodsInvoker<BeforeAttribute>(methodInfo.DeclaringType);
+                    var watch = Stopwatch.StartNew();
+                    object result = methodInfo.Invoke(obj, null);
+                    watch.Stop();
+                    MyTests.Add(new TestStrcuct(methodInfo, isPassed: true, timeConsumed: watch.ElapsedMilliseconds));
+                    MethodsInvoker<AfterAttribute>(methodInfo.DeclaringType);
+                }
+                else
+                {
+                    try
+                    {
+                        MethodsInvoker<BeforeAttribute>(methodInfo.DeclaringType);
+                        var watch = Stopwatch.StartNew();
+                        object result = methodInfo.Invoke(obj, null);
+                        watch.Stop();
+                        MethodsInvoker<AfterAttribute>(methodInfo.DeclaringType);
+                        if (attribute.Expected.Equals(result))
+                        {
+                            MyTests.Add(new TestStrcuct(methodInfo, isPassed: true, expected: attribute.Expected, timeConsumed: watch.ElapsedMilliseconds)); ;
+                        }
+                        else
+                        {
+                            MyTests.Add(new TestStrcuct(methodInfo, expected: attribute.Expected, isFailed: true));
+                        }
+                    }
+                    catch (Exception exception)
+                    {
+                        if (attribute.Expected.Equals(exception.InnerException.GetType()))
+                        {
+                            MyTests.Add(new TestStrcuct(methodInfo, expected: attribute.Expected, isPassed: true));
+                        }
+                        else
+                        {
+                            MyTests.Add(new TestStrcuct(methodInfo, expected: attribute.Expected, isFailed: true));
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Invoking methods with Before and After Attribute
+        /// </summary>
+        public static void MethodsWithAfterAndBeforeAttribute(MethodInfo methodInfo, object obj)
+        {
+            obj = ConstuctorFinder(methodInfo);
+            object result = methodInfo.Invoke(obj, null);
+
+        }
```

Comment:
```suggestion
            object result = methodInfo.Invoke(obj, null);
        }
```
Но зачем нам вообще result?